src/api/api.cpp

```diff
@@ -265,12 +265,17 @@ struct experimental_custom_op
 template <class CustomOp>
 struct custom_operation
 {
+
     template <class Self, class F>
     static auto reflect(Self&, F)
     {
         return pack();
     }
+
+    value attributes() const { return {{"custom_op", true}, {"target", target}}; }
+
     CustomOp op;
+    std::string target = op.runs_on_offload_target() ? "gpu" : "cpu";
```

Comment:
This should be moved into the `attributes` function.

Reply:
Done